include/zephyr/drivers/video-controls.h

```diff
@@ -1,85 +1,162 @@
-/**
- * @file
- *
- * @brief Public APIs for Video.
- */
-
 /*
  * Copyright (c) 2019 Linaro Limited.
+ * Copyright (c) 2024 tinyVision.ai Inc.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef ZEPHYR_INCLUDE_VIDEO_CONTROLS_H_
 #define ZEPHYR_INCLUDE_VIDEO_CONTROLS_H_
 
+/**
+ * @file
+ *
+ * @brief Public APIs for Video.
+ */
+
 /**
  * @brief Video controls
  * @defgroup video_controls Video Controls
  * @ingroup io_interfaces
+ *
+ * The Video control IDs (CIDs) are introduced with the same name as
+ * Linux V4L2 subsystem and under the same class. This facilitates
+ * inter-operability and debugging devices end-to-end across Linux and
+ * Zephyr.
+ *
+ * This list is maintained compatible to the Linux kernel definitions in
+ * @c linux/include/uapi/linux/v4l2-controls.h
+ *
  * @{
  */
 
-#include <zephyr/device.h>
-#include <stddef.h>
-#include <zephyr/kernel.h>
-
-#include <zephyr/types.h>
-
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @name Control classes
+ * @name Base class control IDs
  * @{
  */
-#define VIDEO_CTRL_CLASS_GENERIC 0x00000000 /**< Generic class controls */
-#define VIDEO_CTRL_CLASS_CAMERA  0x00010000 /**< Camera class controls */
-#define VIDEO_CTRL_CLASS_MPEG    0x00020000 /**< MPEG-compression controls */
-#define VIDEO_CTRL_CLASS_JPEG    0x00030000 /**< JPEG-compression controls */
-#define VIDEO_CTRL_CLASS_VENDOR  0xFFFF0000 /**< Vendor-specific class controls */
+#define VIDEO_CID_BASE 0x00980900
+
+/** Amount of perceived light of the image, the luma (Y') value. */
+#define VIDEO_CID_BRIGHTNESS (VIDEO_CID_BASE + 0)
+
+/** Amount of difference between the bright colors and dark colors. */
+#define VIDEO_CID_CONTRAST (VIDEO_CID_BASE + 1)
+
+/** Colorfulness of the image while preserving its brightness */
+#define VIDEO_CID_SATURATION (VIDEO_CID_BASE + 2)
+
+/** Shift in the tint of every colors, clockwise in a RGB color wheel */
+#define VIDEO_CID_HUE (VIDEO_CID_BASE + 3)
+
+/** Amount of time an image sensor is exposed to light, affecting the brightness */
+#define VIDEO_CID_EXPOSURE (VIDEO_CID_BASE + 17)
```

Comment:
I appreciate the effort to make sure all public API is documented but FWIW I find these oddly worded for some reason. I guess that's what happens when the names are already mostly self-explanatory and one doesn't want to literally paraphrase them in the comment. Not blocking though :)

Reply:
Yes, I have same thoughts too and sometimes, the explanation is not totally correct as it is difficult to explain in short words, so it is better to not explain :-).

Reply:
Reading them again, these do not help understanding what the control do.

Better not risk defining them slightly differently than Linux ones.
If there need to be any doc effort, this can go in Linux.

New PR to address these small changes. :)

Reply:
How about a link to this page for reference?
https://www.kernel.org/doc/html/latest/userspace-api/media/v4l/control.html

Reply:
Yes, I think it's a good idea but I am not sure if it's OK in Zephyr to refer to documentation of a different project.